File: 2025-2/sesion-de-clase-11/SoftProgNegocio/BOImpl/ClienteBOImpl.cs
using System.Collections.Generic;
using PUCP.SoftProg.Modelo;
using PUCP.SoftProg.Modelo.Clientes;
using PUCP.SoftProg.Negocio.BO;
using PUCP.SoftProg.Persistencia.DAO.Clientes;
using PUCP.SoftProg.Persistencia.DAOImpl.Clientes;

namespace PUCP.SoftProg.Negocio.BOImpl {
    public class ClienteBOImpl : IClienteBO {
        private readonly IClienteDAO clienteDAO;

        public ClienteBOImpl() {
            this.clienteDAO = new ClienteDAOImpl();
        }

        public void Guardar(Cliente cliente, Estado estado) {
            if (estado == Estado.Nuevo) {
                this.clienteDAO.Crear(cliente);
            }
            else {
                this.clienteDAO.Actualizar(cliente);
            }
        }

        public void Eliminar(int id) {
            this.clienteDAO.Eliminar(id);
        }

        public Cliente Obtener(int id) {
            return this.clienteDAO.Leer(id);
        }

        public List<Cliente> Listar() {
            return this.clienteDAO.LeerTodos();
        }

        public Cliente BuscarPorDni(string dni)
        {
            return this.clienteDAO.BuscarPorDni(dni);
        }
    }
}

File: 2025-2/sesion-de-clase-18/SoftProg/SoftProgNegocio/src/main/java/pe/edu/pucp/inf30/softprog/boimpl/clientes/ClienteBOImpl.java
package pe.edu.pucp.inf30.softprog.boimpl.clientes;

import java.util.List;
import pe.edu.pucp.inf30.softprog.bo.clientes.ClienteBO;
import pe.edu.pucp.inf30.softprog.dao.clientes.ClienteDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.clientes.ClienteDAOImpl;
import pe.edu.pucp.inf30.softprog.modelo.Estado;
import pe.edu.pucp.inf30.softprog.modelo.clientes.Cliente;

/**
 *
 * @author eric
 */
public class ClienteBOImpl implements ClienteBO {
    private final ClienteDAO clienteDao;
    
    public ClienteBOImpl() {
        this.clienteDao = new ClienteDAOImpl();
    }
    
    @Override
    public Cliente buscarPorDni(String dni) {
        return this.clienteDao.buscarPorDni(dni);
    }

    @Override
    public List<Cliente> listar() {
        return this.clienteDao.leerTodos();
    }

    @Override
    public Cliente obtener(int id) {
        return this.clienteDao.leer(id);
    }

    @Override
    public void eliminar(int id) {
        this.clienteDao.eliminar(id);
    }

    @Override
    public void guardar(Cliente modelo, Estado estado) {
        if (estado == Estado.Nuevo) {
            this.clienteDao.crear(modelo);
        }
        else {
            this.clienteDao.actualizar(modelo);
        }
    }
}
